### PR TITLE
[sharktank] Make LLM utils support chunked prefill for included Llama 8b eager quick evaluation tests

### PR DESCRIPTION
**Pull request by sogartar**
Make LlmRunner.prefill use the LLM scheduler mechanism and handle chunked prefill.

Adds pre-submit tests for Llama 3.1 8b f16 and fp8_e4m3fnuz.

For the fp8_e4m3fnuz evaluation tests the accepted tolerance is higher than the
non-chunked test due to different tiling for some ops and brittle error metric
sensitive to outliers. See https://github.com/nod-ai/shark-ai/issues/2494.
We need to refactor this metric in the future.

### CODE REVIEW

File: sharktank/sharktank/utils/llm_utils.py [hunk 1]
```diff
@@ -494,28 +522,45 @@ def run_decode(
...
 
-        prefill_task = PrefillTask(
-            invocation_fn=self._instance.prefill,
-            llm_task_inputs=task_inputs,
-            batch_size=self._prefill_bs,
-            block_stride=self._block_stride,
-        )
-        logits, indices = prefill_task.run(*self._cache)
+        logits = [r[0][0] for r in request_results.values()]
+        indices = [r[0][1] for r in request_results.values()]
```
Comment (archana-ramalingam) [inline sharktank/sharktank/utils/llm_utils.py]: Can be rewritten as `logits, indices = zip(*[(r[0][0], r[0][1]) for r in request_results.values()])`

File: sharktank/sharktank/utils/llm_utils.py [hunk 2]
```diff
@@ -494,28 +522,45 @@ def run_decode(
             cache=self._cache,
         )
 
-    def prefill(self, requests: list[list[int]], page_ids: list[list[int]]):
+    def prefill(
```
Comment (archana-ramalingam) [inline sharktank/sharktank/utils/llm_utils.py]: Can we move the chunk-handling logic into `PrefillTask` so the `LlmRunner` stays minimal. Prefill and decode must do similar minimal steps inside the runner class.

Comment (sogartar) [reply sharktank/sharktank/utils/llm_utils.py]: In the current object model the `LlmRunner` seems the appropriate place to orchestrate this.
The `Scheduler` is concerned with batching and running tasks and the tasks are pieces of work description that should not implement interaction between each other.
The `LlmRunner` already orchestrates this. [Here](https://github.com/nod-ai/shark-ai/blob/7ac5ac6e4c76a7e4e97f59e5f12d6514e14e106f/sharktank/sharktank/utils/llm_utils.py#L373) and [here](https://github.com/nod-ai/shark-ai/blob/7ac5ac6e4c76a7e4e97f59e5f12d6514e14e106f/sharktank/sharktank/utils/llm_utils.py#L432). It breaks up the work into multiple pieces/tasks.

The thing that it did not do is handle the chunking configuration correctly in `prefill`. As it would act as if there is no chunking.

For decode there is no chunking. I think its functionality should not change.

Comment (archana-ramalingam) [reply sharktank/sharktank/utils/llm_utils.py]: Since the `LlmTaskInput` and `PrefillTask` already have chunking related variables, would it fit better in `PrefillTask`, as we can branch in `_prepare_args` and `_process_results`?

Comment (archana-ramalingam) [reply sharktank/sharktank/utils/llm_utils.py]: Yes, `Scheduler` shouldn't do chunking. Linking `LlmDecoder.greedy_decode` and `LlmRunner.prefill` could be a TODO for future PR.

**Review by archana-ramalingam — APPROVED** [review]
